### PR TITLE
kubeadm: unit test for app/apis/kubeadm/env.go

### PR DESCRIPTION
**Pull request by dmmcquay**
Added unit test for the kubeadm/apis/kubeadm package testing functionality of env.go

This PR is part of the ongoing effort to add tests (#35025)

/cc @pires @jbeda

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36176)

### CODE REVIEW

**Comment by pires** [comment]
@k8s-bot ok to test

**Comment by dmmcquay** [comment]
@pires mind rerunning that test? (Jenkins GCI GCE e2e)? Seems like it was just a flake, perhaps?

**Comment by pires** [comment]
@k8s-bot gci gce e2e test this

**Comment by dmmcquay** [comment]
@pires any idea why this is failing?

**Comment by pires** [comment]
@k8s-bot gci gce e2e test this

**Comment by dmmcquay** [comment]
@pires any idea whats going on? Seems strange.

**Comment by pires** [comment]
@k8s-bot test this

**Comment by pires** [comment]
@k8s-bot unit test this

**Comment by dmmcquay** [comment]
@pires mind trying to get the unit tests to pass again?

**Comment by pires** [comment]
@k8s-bot unit test this

**Comment by pires** [comment]
What can we do here @luxas?

**Comment by luxas** [comment]
Hmm, actually we're about to remove this file with the env params so I'm not sure we need tests there...
Anyway, the submit queue is blocked for all non-v1.5 PRs until some time later this month.

**Comment by dmmcquay** [comment]
@luxas @pires So should we push these tests in until those changes happen? Seems like a good idea unless both of you think differently.

**Comment by pires** [comment]
@k8s-bot unit test this

**Comment by pires** [comment]
@k8s-bot test this

**Comment by pires** [comment]
@luxas @mikedanese the fact the failing tests are not related to this PR is beyond me. Can we force the bot to ignore the tests and just be done with this?

**Comment by mikedanese** [comment]
Probably a flake @k8s-bot cvm gce e2e test this

File: cmd/kubeadm/app/apis/kubeadm/env_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		)
+	}
+	if oldEnv.ComponentLoglevel != GlobalEnvParams.ComponentLoglevel {
+		t.Errorf(
+			"failed ComponentLoglevel:\n\texpected: %s\n\t  actual: %s",
+			oldEnv.ComponentLoglevel,
+			GlobalEnvParams.ComponentLoglevel,
+		)
+	}
+	defer func() { GlobalEnvParams = oldEnv }()
```
Comment (mikedanese) [inline cmd/kubeadm/app/apis/kubeadm/env_test.go]: why are you deferring at the end of the function call? It's not doing anything.

**Comment by mikedanese** [comment]
All this test does is check that SetEnvParams() returns the same map twice? This is equivalent to checking reflect.DeepEqual(GlobalEnv, SetEnvParams()) (actually that is much better because it's much shorter, doesn't require rewriting a global variable and doesn't need modification if we add new fields to GlobalEnv). Even that I don't think is useful.

**Comment by rmmh** [comment]
@k8s-bot test this issue kubernetes/test-infra#1253

**Comment by dmmcquay** [comment]
@mikedanese do you suggest just closing this PR then? I was just working on trying to get as much coverage as possible throughout the kubeadm code base. If you don't see any value in verifying SetEnvParams() we should just close it.

**Comment by luxas** [comment]
Also, we're gonna remove this code pretty soon...

**Comment by luxas** [comment]
@dmmcquay Do you mind if I close this then?
I appreciate the other PRs and hope they'll merge soon now when the SQ has opened

**Comment by dmmcquay** [comment]
@luxas sure, I'll close this PR.

**Pull request closed without merging**